tests/core/test_actions.py

```diff
@@ -2732,3 +2732,76 @@ async def test_action_extract_slots_emits_necessary_slot_set_events(
         assert events[0].value == value_to_set
     else:
         assert len(events) == 0
+
+
+async def test_action_extract_slots_priority_of_slot_mappings():
+    slot_name = "location_slot"
+    entity_name = "location"
+    entity_value = "Berlin"
+
+    domain_yaml = textwrap.dedent(
+        f"""
+        version: "{LATEST_TRAINING_DATA_FORMAT_VERSION}"
+
+        intents:
+        - greet
+        - goodbye
+        - affirm
+        - deny
+        - mood_great
+        - mood_unhappy
+        - bot_challenge
+        - inform
+
+        entities:
+        - {entity_name}
+
+        slots:
+          {slot_name}:
+            type: text
+            influence_conversation: false
+            mappings:
+            - type: from_entity
+              entity: {entity_name}
+            - type: from_intent
+              value: 42
+              intent: inform
+
+        forms:
+          test_form:
+            required_slots:
+            - {slot_name}
+
+        responses:
+            utter_test:
+                - condition:
+                  - type: slot
+                    name: {slot_name}
+                    value: null
+                  text: "location slot is null"
+                - text: "location slot has value: {{slot_name}}"
+
```

Comment:
This can be simplified since it isn't required for the test (afaict)

Reply:
True, I'll amend that, I initially copied entirely the files from the reproducing repo.